subsys/net/ip/tcp.c

```diff
@@ -1923,6 +1923,19 @@ static struct tcp *tcp_conn_new(struct net_pkt *pkt)
 		goto err;
 	}
 
+	/* Set the local port in the context object after the net_context_bind()
+	 * call in order to avoid port reuse problems (as REUSEADDR / REUSEPORT
```

Comment:
`tcp_conn_new()` is used to create a new TCP client connection during `accept()`. During debugging the `TIME_WAIT` stuff, I encountered that the remote address and port, as well as the local address, are properly written to the related `net_context` object (remote address and port in line 1901, local address within the call to`net_context_bind`). However, the local port is missing. 

In case this connection is closed and entered the `TIME_WAIT` state, the missing local port in the net_context object resulted in this object not properly being found during a `bind()` call to the same local address / port combination.

However, we cannot add the local port to the `local_addr' variable prior to the call to `net_context_bind`, as this call will then fail because the SO_REUSEPORT option is not enabled on both net_context options. Therefore, we just add the port afterward to make sure the net_context objects holds the information.

Reply:
You explanation is good, could you add most of it to the comment in the code. The existing comment is a bit too vague.